impl/maven-impl/src/main/java/org/apache/maven/impl/PathSelector.java

```diff
@@ -528,12 +560,20 @@ private static PathMatcher[] matchers(final FileSystem fs, final String[] patter
         return matchers;
     }
 
+    /**
+     * {@return whether there is no include or exclude filters}.
+     * In such case, this {@code PathSelector} instance should be ignored.
+     */
+    public boolean isEmpty() {
+        return (includes.length | excludes.length) == 0;
```

Comment:
I'm not sure this is right, and even if it is, it's hard to think about bitwise operations. Prefer a simple

if includes.length == 0 && excludes.length == 0